src/Microsoft.SqlTools.Migration/MigrationService.cs

```diff
@@ -1088,43 +1088,72 @@ internal static List<AzureSqlSkuCategory> GetEligibleSkuCategories(string target
                     break;
 
                 case "AzureSqlManagedInstance":
-                    // Gen5 BC/GP MI
+                    // Gen5 BC/Next-GenGP/GP MI
                     eligibleSkuCategories.Add(new AzureSqlSkuPaaSCategory(
                                                     AzureSqlTargetPlatform.AzureSqlManagedInstance,
                                                     AzureSqlPurchasingModel.vCore,
                                                     AzureSqlPaaSServiceTier.BusinessCritical,
                                                     ComputeTier.Provisioned,
                                                     AzureSqlPaaSHardwareType.Gen5));
+                    if (recommendNextGen)
+                    {
+                        eligibleSkuCategories.Add(new AzureSqlSkuPaaSCategory(
+                                                        AzureSqlTargetPlatform.AzureSqlManagedInstance,
```

Comment:
nit : we can combine 3 if statements into single if ?

Reply:
There is an order to be followed for categories. To maintain the order it is added differently.